src/psyclone/dynamo0p3.py

```diff
@@ -3271,17 +3275,23 @@ def _invoke_declarations(self, parent):
 
         for intent in FORTRAN_INTENT_NAMES:
             if self._real_scalars[intent]:
-                parent.add(DeclGen(parent, datatype="real",
-                                   kind=api_config.default_kind["real"],
-                                   entity_decls=self._real_scalars[intent],
+                arg_list = [arg.declaration_name for arg in
+                            self._real_scalars[intent]]
+                dtype = "real"
+                parent.add(DeclGen(parent, datatype=dtype,
+                                   kind=api_config.default_kind[dtype],
+                                   entity_decls=arg_list,
                                    intent=intent))
 
         for intent in FORTRAN_INTENT_NAMES:
             if self._int_scalars[intent]:
+                arg_list = [arg.declaration_name for arg in
```

Comment:
Since it's only the names we use in these local lists (`self._real_scalars` and `self._int_scalars`) I suggest we ammend the code that sets them up so that it still only stores the names and not the Arg objects. We could also name those lists appropriately by adding the `_names` suffix, e.g. `self._real_scalar_names`.

Reply:
Ok, I amended the `__init__` method to store only names and not the objects and I also renamed those dictionaries as suggested. Please note that this required a  `for` loop over Fotran intents for an invoke scalar arguments (similar to the one for the kernels).

Reply:
Ok, I amended the `__init__` method to store only names and not the objects and I also renamed those dictionaries as suggested. Please note that this required a  `for` loop over Fotran intents for an invoke scalar arguments (similar to the one for the kernels).